lib/xboxrt/Makefile

```diff
@@ -0,0 +1,17 @@
+SRCS += \
+	$(NXDK_DIR)/lib/xboxrt/libc_extensions/debug.c \
+	$(NXDK_DIR)/lib/xboxrt/libc_extensions/stat.c \
```

Comment:
*Sidenote: This file is currently entirely dead code (disabled by `#if 0`). Do we intend to revive it?*

Reply:
Not really sure. Afaik Win32 has it, so it's probably a good idea to implement this properly, and some libc++ filesystem code wants to have that, too (although that code in libc++ is broken anyway atm). Should I eventually do it, I'd to it from scratch though and on top of winapi functions.
Afaik we only keep `stat.c` around because it forms a pair with `stat.h` and that one is required by our usb code.

Reply:
The MS extensions (aiming for POSIX compatibility probably) are usually prefixed, so it would be `_stat`. The UNIX and POSIX variants aren't usually prefixed: `stat`. Win32 would probably use `GetFileAttributes` instead (also see #161).

I think it's more interesting to fix on winapi support (which has been going quite well), and we could probably use old mingw wrappers in the future (the new ones seem to use `_stat32` from the MS CRT, but I believe they used to use Winapi directly).

stat.h can definitely stay for now (at least it declares 1 type). I'm still not sure what to do with stat.c - for this PR, I'm fine with just keeping it around.